numpy/core/arrayprint.py

```diff
@@ -1035,7 +1047,10 @@ def format_float_positional(x, precision=None, unique=True,
     >>> np.format_float_positional(np.float16(0.3), unique=False, precision=10)
     '0.3000488281'
     """
-    precision = -1 if precision is None else precision
+    if precision is None:
+        precision = -1
+    elif precision < 0:
+        raise ValueError("precision must be >= 0")
```

Comment:
Why are we now forbidding `-1` passed directly? If we do it here, shouldn't you do it below too?

Reply:
I consider the `-1` to be an ugly internal detail of the Python-C interface for float printing, forced by the behavior of optional args in `PyArgParseTuple`, which I want to hide. It's much clearer if the user should supply `None` if they mean unlimited precision. (Other `multiarraymodule.c` functions also have this `-1` quirk, if I remember correctly).

Where do you mean by "below"? I thought I caught all the places the user can supply precision.

Reply:
The padding arguments on the next line

Reply:
`pad_left = -1 if pad_left is None else pad_left`, for example.

Perhaps a simple converter function in C would be worthwhile here?
```
int positive_or_none(PyObject *obj, void* ctx) {
    int *result = ctx;
    if (obj == Py_None) {
        *result = -1;
        return 0;
    }
    // convert to int, check sign
}
```